cmd/bundle/generate/utils.go

```diff
@@ -103,8 +119,19 @@ func (n *downloader) markNotebookForDownload(ctx context.Context, notebookPath *
 
 	ext := notebook.GetExtensionByLanguage(info)
 
-	filename := path.Base(*notebookPath) + ext
-	targetPath := filepath.Join(n.sourceDir, filename)
+	if n.basePath == "" {
```

Comment:
Do we need to modify `markNotebookForDownload`? Seems like only `markFileForDownload` is called for apps generate.

Reply:
For apps - yes, but generally it makes sense to make this improvement for all types of files we donwload

Reply:
But we only call `markDirectoryForDownload` in apps. We also don't call `markNotebookForDownload` in `markDirectoryForDownload`.

Is there still an improvement to be had?

Reply:
Right now we don't need, this is more for consistency between the functions, so all of the respect the basePath